akka-persistence-typed/src/main/scala/akka/persistence/typed/internal/EventSourcedBehaviorImpl.scala

```diff
@@ -199,7 +199,7 @@ private[akka] final case class EventSourcedBehaviorImpl[Command, Event, State](
                 case res: SnapshotProtocol.Response          => InternalProtocol.SnapshotterResponse(res)
                 case RecoveryPermitter.RecoveryPermitGranted => InternalProtocol.RecoveryPermitGranted
                 case internal: InternalProtocol              => internal // such as RecoveryTickEvent
-                case cmd: Command @unchecked                 => InternalProtocol.IncomingCommand(cmd)
+                case cmd                                     => InternalProtocol.IncomingCommand(cmd.asInstanceOf[Command])
```

Comment:
@dwijnand it would have been nice if there was an annotation, like `nowarn` to tell the compiler that we know better? is that already possible or do you have any plans for it?

Reply:
That's `(msg: @unchecked) match {`, if you don't want to use `nowarn`.

Reply:
ok, so the `unchecked` must be at the top instead of the match case.

can `@nowarn` be at the same place?

```
(msg: @nowarn) match {
```

@johanandren what do you think about that? might make the intention more clear where we give up on exhaustiveness check?

Reply:
I think in most of these cases the changes, even if annoying, are more clear than a `@nowarn`, I think you could mostly categorize the cases into 4 different:

 * `OptionVal` - more clear IMO to just have a fallback for the "other" case
 * Akka typed uses of receive but really meaning receivePartial - does not make sense with `@nowarn` for those
 * Methods (Testkit `fishForMessages`) where we kind of intended `PartialFunction` but kind of accidentally took a total function - better to add that partial signature
 * last case fallbacks like this one, where we know the type, but the compiler doesn't - I think this is the one closest to a case where `@nowarn` makes sense, but personally I anyway like the catchall and explicit typecast better, even than `:T @unchecked =>` and the syntax overhead is pretty much the same

Reply:
> can `@nowarn` be at the same place?
> 
> ```
> (msg: @nowarn) match {
> ```

I think so.